### PR TITLE
feat: Amplify geofence control

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Merging Geofence Control into main branch

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,14 +36,19 @@
   "homepage": "https://github.com/aws-amplify/maplibre-gl-js-amplify#readme",
   "devDependencies": {
     "@aws-amplify/core": "^4.3.1",
-    "@aws-amplify/geo": "^1.1.1",
+    "@aws-amplify/geo": "geo",
```
Comment (Amplifiyer) [inline package.json:39]: Should we pin a version that we know works with this release? There could be some breaking changes before GA in amplify-js library

Comment (thaddmt) [reply package.json:39]: I'll update to whatever the current working version is

File: src/AmplifyGeofenceControl/AmplifyMapboxDraw.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  getPolygonFeatureFromBounds,
+  getCircleFeatureFromCoords,
+} from "../geofenceUtils";
+import {
+  GEOFENCE_BORDER_COLOR,
+  GEOFENCE_COLOR,
+  GEOFENCE_VERTEX_COLOR,
+} from "../constants";
+
+export class AmplifyMapboxDraw {
```
Comment (Amplifiyer) [inline src/AmplifyGeofenceControl/AmplifyMapboxDraw.ts:20]: Should we just call `AmplifyMapDraw`?

Comment (thaddmt) [reply src/AmplifyGeofenceControl/AmplifyMapboxDraw.ts:20]: Will change

**Pull request merged**